logdna/index.js

```diff
@@ -106,13 +107,21 @@ const sendLine = (payload, config, callback) => {
     // Flush the Log
     asyncRetry({
         times: MAX_REQUEST_RETRIES
-        , interval: REQUEST_RETRY_INTERVAL_MS
-        , errorFilter: (err) => {
-            return err.code === 'ETIMEDOUT' || err.code === 'ESOCKETTIMEDOUT';
+        , interval: (retryCount) => {
+            return REQUEST_RETRY_INTERVAL_MS * Math.pow(2, retryCount);
+        }
+        , errorFilter: (errCode) => {
+            console.log(errCode);
+            return DEFAULT_HTTP_ERRORS.includes(errCode);
         }
     }, (reqCallback) => {
         return request(options, (error, response, body) => {
-            if (error) return reqCallback(error);
+            if (error) {
+                return reqCallback(error.code);
+            } else if (response.statusCode >= INTERNAL_SERVER_ERROR) {
```

Comment:
Remove `else` since it's already `return`ing in the previous one - have flat `if`s